witchslabyrinth/WitchsLabyrinthGame
# Disable cat control until koi puzzle

## Changes

Player can no longer swap with/command friend actor until completing the Liars and Pattern puzzles
(more specifically, not until walking through the backyard doors)

Also changed the PlayerController in scene with a prefab instance, to make future changes easier

#### Files Changed:
- `Ukiyo-e Environment.unity` (easily repeatable)
- `PlayerController.prefab`
- `PlayerController.cs`
- `PlayerInteractionController.cs`
- `ZodiacPuzzle.cs`
- `PatternPuzzle.cs`
- `DetectLiarsAndPatternPuzzleCompleted.cs`
- `DialogueCommands.cs`

## Testing
- Ensure you can't swap before completing the Liars and Pattern puzzles
- Ensure you can't swap during any cutscenes, or any other places that shouldn't allow it

## Known Bugs

### The player can swap to (and gain control of) the other actor during the cutscene leading to the Portal Puzzle

![image](https://user-images.githubusercontent.com/23004127/82847656-5e84f200-9ebd-11ea-8077-05379005c8d0.png)

![image](https://user-images.githubusercontent.com/23004127/82847678-76f50c80-9ebd-11ea-9bf5-e318043e9c08.png)

### Trying to swap back to the player actor will break the camera:
![image](https://user-images.githubusercontent.com/23004127/82847713-a3a92400-9ebd-11ea-9792-e93ca40da1bd.png)
